### PR TITLE
Treat comments beginning with four slashes as regular line comments

### PR DESCRIPTION
**Pull request by adamrk**
Addresses https://github.com/rust-analyzer/rust-analyzer/issues/4040

### CODE REVIEW

**Comment by matklad** [comment]
lgtm, but tests are failing?

bors d+

File: crates/ra_syntax/src/ast/tokens.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub enum CommentPlacement {
 const COMMENT_PREFIX_TO_KIND: &[(&str, CommentKind)] = {
     use {CommentPlacement::*, CommentShape::*};
     &[
+        ("////", CommentKind { shape: Line, doc: None }),
```
Comment (kjeremy) [inline crates/ra_syntax/src/ast/tokens.rs:51]: Is `////` specifically called out in rust? If not I think the code that returns the text should be changes instead.

Comment (adamrk) [reply crates/ra_syntax/src/ast/tokens.rs:56]: Yeah, it explicitly says a line comment begins with either `//` or `////`: https://doc.rust-lang.org/reference/comments.html
image

Comment (kjeremy) [reply crates/ra_syntax/src/ast/tokens.rs:56]: TIL!

Comment (adamrk) [reply crates/ra_syntax/src/ast/tokens.rs:56]: Me too!

**Pull request merged**